src/utils/vc/credentials.js

```diff
@@ -0,0 +1,245 @@
+  // Ensure credential status is valid, if exists
+  if ('credentialStatus' in credential) {
+    if (!credential.credentialStatus.id) {
+      throw new Error('"credentialStatus" must include an id.');
+    }
+    if (!credential.credentialStatus.type) {
+      throw new Error('"credentialStatus" must include a type.');
+    }
+  }
+
+  // Ensure expiration date is valid, if exists
+  if ('expirationDate' in credential) {
+    ensureValidDatetime(credential.expirationDate);
+  }
+}
+
+/**
+ * @param {object} credential - An object that could be a VerifiableCredential.
+ * @throws {Error}
+ */
+export function checkCredential(credential) {
+  checkCredentialRequired(credential);
+  checkCredentialOptional(credential);
+  checkCredentialJSONLD(credential);
+}
+
+/**
+* @typedef {object} VerifiableParams The Options to verify credentials and presentations.
+* @property {string} [challenge] - proof challenge Required.
+* @property {string} [domain] - proof domain (optional)
+* @property {string} [controller] - controller (optional)
+* @property {DIDResolver} [resolver] - Resolver to resolve the issuer DID (optional)
+* @property {Boolean} [unsignedPresentation] - Whether to verify the proof or not
+* @property {Boolean} [compactProof] - Whether to compact the JSON-LD or not.
+* @property {Boolean} [forceRevocationCheck] - Whether to force revocation check or not.
+* Warning, setting forceRevocationCheck to false can allow false positives when verifying revocable credentials.
+* @property {object} [purpose] - A purpose other than the default CredentialIssuancePurpose
+* @property {object} [revocationApi] - An object representing a map. "revocation type -> revocation API". The API is used to check
+* revocation status. For now, the object specifies the type as key and the value as the API, but the structure can change
+* as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
+* @property {object} [schemaApi] - An object representing a map. "schema type -> schema API". The API is used to get
+* a schema doc. For now, the object specifies the type as key and the value as the API, but the structure can change
+* as we support more APIs there are more details associated with each API. Only Dock is supported as of now.
+* @property {object} [documentLoader] - A document loader, can be null and use the default
+*/
+
+/**
+ * Verify a Verifiable Credential. Returns the verification status and error in an object
+ * @param {object} [credential] The VCDM Credential
+ * @param {VerifiableParams} options Verify parameters, this object is passed down to jsonld-signatures calls
+ * @return {Promise<object>} verification result. The returned object will have a key `verified` which is true if the
+ * credential is valid and not revoked and false otherwise. The `error` will describe the error if any.
+ */
+// TODO: perform verification on the expanded JSON-LD credential
```

Comment:
i have moved this TODO into another issue as itll effect too much logic for this PR, imo. better to address separately